spec/rspec/matchers/composable_spec.rb

```diff
@@ -95,6 +95,26 @@ module Matchers
           expect(with_matchers_cloned([stdout]).first).to equal(stdout)
         end
       end
+
+      describe "when an unexpected call stack jump occurs" do
+        RSpec::Matchers.define :cause_call_stack_jump do
+          supports_block_expectations
+
+          match do |block|
+            block.call
+            false
+          rescue Exception
+            true
+          end
+        end
+
+        it "will not raise an index or key error" do
```

Comment:
It's generally better to positively express what the behavior _is_ rather than negatively express what the behavior _is not_.  The index or key error was entirely due to an internal implementation detail (our use of a hash and `Hash#fetch`).  Instead, I think it's better to say something like:

```
it "provides a clear directions to the user about the need for `expects_call_stack_jump?`" do
  # ...
end
```